src/System.Private.Xml.Linq/tests/TreeManipulation/XAttributeEnumRemove.cs

```diff
@@ -245,9 +245,9 @@ private void VerifyDeleteAttributes(IEnumerable<XAttribute> allAttributes)
             var expectedAttrsForParent = new Dictionary<XElement, List<ExpectedValue>>();
             foreach (XElement p in parents)
             {
-                if (p != null && !expectedAttrsForParent.ContainsKey(p))
+                if (p != null)
                 {
-                    expectedAttrsForParent.Add(p, p.Attributes().Except(copyAllAttributes.Where(x => x != null)).Select(a => new ExpectedValue(true, a)).ToList());
+                    expectedAttrsForParent.TryAdd(p, p.Attributes().Except(copyAllAttributes.Where(x => x != null)).Select(a => new ExpectedValue(true, a)).ToList());
```

Comment:
This change needs to be reverted.  It's adding a lot of expense even if the key is already in the dictionary.